recorder.go

```diff
@@ -104,3 +105,72 @@ func (p *Printer) Record(loc *Location, op Operation, stats *Stats) error {
 	p.lastValue = time.Now()
 	return nil
 }
+
+// Full Recorder records all of the evaluations that occur during an optimization
```

Comment:
Do you have a justification why this should be in optimize? The functionality seems quite specific to fit general needs.